src/hooks/useGovernance.ts

```diff
@@ -0,0 +1,167 @@
+import { newKit } from '@celo/contractkit';
+import {
+  GovernanceWrapper,
+  ProposalRecord,
+  ProposalStage,
+} from '@celo/contractkit/lib/wrappers/Governance';
+import { useCelo } from '@celo/react-celo';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { getRawGithubUrl, ParsedYAML, parsedYAMLFromMarkdown } from 'src/utils/proposals';
+
+export type Proposal = ProposalRecord & {
+  proposalID: string;
+  markdown: string;
+  parsedYAML: ParsedYAML | null;
+};
+
+export function useGovernance() {
+  const { network } = useCelo();
+  const contractKit = useMemo(() => newKit(network.rpcUrl), [network]);
+  const [governance, setGovernance] = useState<GovernanceWrapper>();
+
+  const [queue, setQueue] = useState<null | Proposal[]>(null);
+  const [approval, setApproval] = useState<null | Proposal[]>(null);
+  const [referendum, setReferendum] = useState<null | Proposal[]>(null);
+  const [execution, setExecution] = useState<null | Proposal[]>(null);
+  const [expiration, setExpiration] = useState<null | Proposal[]>(null);
+  const allProposals = useMemo(
+    () =>
+      (
+        [queue, execution, expiration, approval, referendum].flat().filter(Boolean) as Proposal[]
+      ).sort((a, b) => b.metadata.timestamp.minus(a.metadata.timestamp).toNumber()),
+    [queue, execution, expiration, approval, referendum]
+  );
+  const [error, setError] = useState<null | Error>(null);
+
+  useEffect(() => void contractKit.contracts.getGovernance().then(setGovernance), [contractKit]);
```

Comment:
should e maybe bring in something like https://www.npmjs.com/package//use-async-effect

Reply:
agreed, maybe in a later pr